crates/oxc_linter/src/rules/import/no_commonjs.rs

```diff
@@ -214,6 +213,10 @@ impl Rule for NoCommonjs {
                     return;
                 }
 
+                if ctx.scopes().find_binding(ctx.scopes().root_scope_id(), "require").is_some() {
```

Comment:
what about:
```
function foo() { 
    const require = () => {};
    require('foo')
}
```

this should pass but i think it'd be reported since `require` isn't in the root scope?

probably want to use https://github.com/oxc-project/oxc/blob/9a6a2f99c60638be1904bba77c991824bbf3549e/crates/oxc_semantic/src/symbol.rs#L275

Reply:
good catch, looks like it's not catched by original `import/no-commonjs` rule but original `typescript/no-require-imports` catch it